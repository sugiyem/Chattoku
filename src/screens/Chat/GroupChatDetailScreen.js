import React, { useEffect, useState } from "react";
import { Alert, LogBox, StyleSheet, View } from "react-native";
import { Button, ButtonText } from "../../styles/GeneralStyles";
import { firebase } from "../../services/Firebase/Config";
import { chatType } from "../../constants/Chat";
import { fetchGroupChatMessages } from "../../services/Chat/FetchChatMessages";
import FetchUserInfo from "../../services/Profile/FetchUserInfo";
import ChatSections from "../../components/Chat/ChatSections";

const initialState = {
  username: "",
  img: ""
};

// Ignore warnings from Animated (Because of Gifted Chat)
LogBox.ignoreLogs(["Animated"]);

const GroupChatDetailScreen = ({ navigation, route }) => {
  const [userInfo, setUserInfo] = useState(initialState);
  const [messages, setMessages] = useState([]);

  const userID = firebase.auth().currentUser.uid;
  const groupID = route.params.groupID;
  const groupName = route.params.groupName;

  useEffect(() => {
    return FetchUserInfo({
      onSuccesfulFetch: (data) => {
        setUserInfo(data);
      },
      onFailure: (error) => {
        Alert.alert(error.message);
      }
    });
  }, []);

  useEffect(() => {
    return fetchGroupChatMessages({
      groupID: groupID,
      onSuccess: (data) => {
        setMessages(data);
      },
      onFailure: (error) => {
        Alert.alert(error.message);
      }
    });
  }, []);

  return (
    <View style={styles.container}>
      <Button onPress={() => navigation.replace("GroupChatList")}>
        <ButtonText>
          {"Currently in a group chat with " +
            groupName +
            ".\n Click here to go to the group chat list"}
        </ButtonText>
      </Button>

      <ChatSections
        type={chatType.GROUP_CHAT}
        userData={{ ...userInfo, id: userID }}
        receiverID={groupID}
        messages={messages}
        updateMessages={setMessages}
      />
    </View>
  );
};

export default GroupChatDetailScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "darkcyan",
    padding: 5
  }
});
