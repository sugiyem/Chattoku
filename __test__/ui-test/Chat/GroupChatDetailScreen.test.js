import React from "react";
import renderer from "react-test-renderer";
import GroupChatDetailScreen from "../../../src/screens/Chat/GroupChatDetailScreen";

jest.mock("../../../src/services/Friend/HandleBlockedUser");
jest.mock("../../../src/services/Friend/FetchFriendStatus");
jest.mock("../../../src/services/Friend/FriendshipStatus");
jest.mock("../../../src/services/Friend/HandleGroup");
jest.mock("../../../src/services/Chat/FetchChatMessages");
jest.mock("../../../src/services/Profile/FetchUserInfo");
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: (initial) => [initial, jest.fn()]
}));

const mockNavigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  push: jest.fn(),
  replace: jest.fn()
};

const mockRoute = {
  params: {
    groupData: {
      id: "group-1",
      name: "Weebs",
      description: "For weeb only",
      img: "w-img"
    }
  }
};

describe("Test Group Chat Page UI", () => {
  it("Renders correctly", () => {
    const tree = renderer
      .create(
        <GroupChatDetailScreen navigation={mockNavigation} route={mockRoute} />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
