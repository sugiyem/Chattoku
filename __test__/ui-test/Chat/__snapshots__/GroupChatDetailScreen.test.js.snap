// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Group Chat Page UI Renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "darkcyan",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 5,
        "paddingLeft": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "cyan",
          "borderColor": "blue",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "marginBottom": 10,
          "paddingVertical": 10,
          "width": 740,
        },
      ]
    }
  >
    <View
      size="1"
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="backIcon"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      size="2"
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 2,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "w-img",
          }
        }
        style={
          Array [
            Object {
              "borderColor": "black",
              "borderRadius": 80,
              "borderWidth": 1,
              "height": 80,
              "width": 80,
            },
          ]
        }
        testID="profileImage"
      />
    </View>
    <View
      size="2"
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 2,
            "flexShrink": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Gill Sans",
              "fontSize": 18,
              "fontWeight": "600",
            },
          ]
        }
        testID="name"
      >
        Weebs
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontSize": 14,
            },
          ]
        }
        testID="info"
      >
        For weeb only
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 3,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        isSmall={true}
        style={
          Array [
            Object {
              "marginHorizontal": 5,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            testID="groupPageIcon"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
        <Text
          color="navy"
          isSmall={true}
          style={
            Array [
              Object {
                "color": "navy",
                "fontSize": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Group Info
        </Text>
      </View>
      <View
        isSmall={true}
        style={
          Array [
            Object {
              "marginHorizontal": 5,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "overflow": "hidden",
            }
          }
        >
          <View
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "justifyContent": "center",
                }
              }
            >
              <Text />
            </View>
          </View>
        </View>
        <Text
          color="#c10015"
          isSmall={true}
          style={
            Array [
              Object {
                "color": "#c10015",
                "fontSize": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          Leave
        </Text>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
